src/tests/core/engine/processors/kitchen_simulator_test.py

```diff
@@ -0,0 +1,40 @@
+from src.core.engine.processors.kitchen_simulator import KitchenSimulator
+from src.tests.utils.fixtures.app_processor_config_fixture import (
+    build_app_processor_config,
+)
+from src.tests.utils.fixtures.chef_fixture import build_chef
+from src.tests.utils.fixtures.order_fixture import build_order
+from src.constants.order_status import OrderStatus
+from unittest import mock
+import unittest
+
+
+class KitchenSimulatorTest(unittest.TestCase):
+    def test_assign_orders_to_available_chefs(self):
+
+        chef_1 = build_chef(chef_id=1)
+        order_1 = build_order(order_id=1, status=OrderStatus.NEW_ORDER)
+        app_engine_config = build_app_processor_config()
+        kitchen_simulator = KitchenSimulator(app_engine_config)
+        kitchen_simulator.order_manager = mock.Mock()
+        kitchen_simulator.chef_controller = mock.Mock()
```

Comment:
Could you change this mock to a setup function instead of doing this for each test? Also can you add test cases where you have both processes cancel and in process?

Reply:
done